torch/distributions/transforms.py

```diff
@@ -967,6 +968,23 @@ def _inverse(self, y):
         return y.tril(-1) + y.diagonal(dim1=-2, dim2=-1).log().diag_embed()
 
 
+class PositiveDefiniteTransform(LowerCholeskyTransform):
+    """
+    Transform from unconstrained matrices to positive-definite matrices.
+    """
+    codomain = constraints.positive_definite  # type: ignore[assignment]
+
+    def __eq__(self, other):
+        return isinstance(other, PositiveDefiniteTransform)
+
+    def _call(self, x):
+        x = super()._call(x)
+        return x @ x.mT
+
+    def _inverse(self, y):
+        y = torch.linalg.cholesky(y)
```

Comment:
I think you can optimize the ops with [torch.linalg.cholesky_ex](https://pytorch.org/docs/stable/generated/torch.linalg.cholesky_ex.html) if you want.

Reply:
If you know for a fact that Cholesky is going to succeed (which may not be the case if the matrix has very small eigenvalues) then I'd very much suggest to use `linalg.cholesky_ex`. Otherwise, if you want to be careful (at the expense of  one synchronisation host-device, `linalg.cholesky` will do just fine.

Reply:
Thank you for the analysis.

Reply:
I'd say, given that this is a linear algebra algorithm that has a fair amount of numerical errors, just use `linalg.cholesky`. If people then complain about the speed and file and issue, we can always reconsider this choice.